Model/IdealPaymentMethod.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Model;
+
+use Wirecard\PaymentSdk\Config\PaymentMethodConfig;
+use Wirecard\PaymentSdk\Entity\IdealBic;
+use Wirecard\PaymentSdk\Transaction\IdealTransaction;
+use Wirecard\PaymentSdk\Transaction\Transaction;
+
+use Wirecard\Oxid\Core\Helper;
+
+use OxidEsales\Eshop\Core\Registry;
+
+/**
+ * Payment method implementation for iDEAL
+ *
+ * @since 1.2.0
+ */
+class IdealPaymentMethod extends PaymentMethod
+{
+    /**
+     * @inheritdoc
+     *
+     * @since 1.2.0
+     */
+    protected static $_sName = "ideal";
+
+    /**
+     *
+     * @since 1.2.0
+     */
+    private $_aBankOptions = [];
+
+    /**
+     * IdealPaymentMethod constructor.
+     *
+     * @since 1.2.0
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->_aBankOptions = $this->getBanks();
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return Config
+     *
+     * @since 1.2.0
+     */
+    public function getConfig()
+    {
+        $oConfig = parent::getConfig();
+
+        $oPaymentMethodConfig = new PaymentMethodConfig(
+            IdealTransaction::NAME,
+            $this->_oPayment->oxpayments__wdoxidee_maid->value,
+            $this->_oPayment->oxpayments__wdoxidee_secret->value
+        );
+
+        $oConfig->add($oPaymentMethodConfig);
+
+        return $oConfig;
+    }
+
+    /**
+     * Get the current transaction to be processed
+     *
+     * @return Transaction
+     *
+     * @since 1.2.0
+     */
+    public function getTransaction()
+    {
+        $oTransaction = new IdealTransaction();
+
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+
+        $oTransaction->setBic($this->_aBankOptions[$aDynvalues['bank']]);
```

Comment:
I think this should be done in `addMandatoryTransactionData`.

Reply:
Moved to addMandatoryTransactionData. :+1: